processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -27,45 +28,97 @@ import (
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-func TestJobE2E(t *testing.T) {
-	f, err := os.Open("./testdata/trace.json")
-	assert.NoError(t, err)
+func TestTraceE2E(t *testing.T) {
+
+	tcs := []struct {
+		name  string
+		attrs map[string]string
+	}{
+		{
+			name: "job",
+			attrs: map[string]string{
+				"k8s.namespace.name": "default",
+				"k8s.job.name":       "test-telemetrygen-job",
+			},
+		},
+		{
+			name: "statefulset",
+			attrs: map[string]string{
+				"k8s.namespace.name":   "default",
+				"k8s.statefulset.name": "test-telemetrygen-statefulset",
```

Comment:
Can please you cover all attributes that are expected to be set by the k8sattributes processor by default? `k8s.pod.name` must be predictable for the statefulset. If we cannot predict some attribute values, can we check their presence?

Reply:
I enriched the compare mode into "equal", "regex" and "exist".

- equal for exactly matching
- regex for matching by regex pattern
- exist for checking if the key exists

Reply:
Thank you @fatsheep9146 . It's super comprehensive coverage!